docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,250 @@
+same Image association.
+
+The data of an Image::Revision is stored between an Image::FileRevision and an
+Image::MetadataRevision. These two objects differ by the fact that any change to
+Image::FileRevision requires changes to the resultant Asset Manager files (such
+as crop dimensions), whereas Image::MetadataRevision stores accompanying data
+that doesn't affect the Asset Manager files (such as alt text).
+
+Image::FileRevision is associated with an ActiveStorage::Blob object that is
+responsible for managing the storage of the source file. It also has a one to
+many association with Image::Asset, each of these represent resulting files that
+are uploaded to Asset Manager. The Image::Asset model stores the URL to the
+Asset Manager file and what state the file is on Asset Manager.
+
+### Timeline
+
+The TimelineEntry model represents an event that should be shown to a user as
+part of a visual timeline of a document's history. It stores the event which
+occurred and associations to relevant models. The sole purpose of this model
+is for representing a timeline to users visually and not for wider historical
+uses outside of the timeline view.
+
+The TimelineEntry model should not store data which could not be
+derived from other aspects of a document. This is to maintain
+flexibility for timeline to be an evolving and iterated part of the
+application. Previously models such as Removal and Withdrawal were associated
+directly with a TimelineEntry which led to using this model outside of
+displaying a timeline. These are now suggested to be associated with a Status
+model.
+
+At the time of writing it wasn't yet determined what the
+timeline would show, and therefore it wasn't clear exactly how
+best to model them. Because of this TimelineEntry will be modelled in a
+speculative way with a number of references to relevant data.
+
+### Topics
+
+No data related to topics (otherwise known as GOV.UK taxonomy) is intended to
+be stored in Content Publisher at the current point in time. This is due to
+topics being accessed and edited by directly interacting with the Publishing
+API. Other applications, notably Content Tagger, can also edit topics which has
+the consequence that the Publishing API is the source of truth for this data
+rather than Content Publisher.
+
+This inconsistency makes it difficult to store the history of topics in a
+reliable way. Thus, until needs are determined to store this data, Content
+Publisher will continue to query the Publishing API rather than storing this
+data.
+
+### Overview
+
+The various aspects of this decision can be collated together to be represented
+in the following, somewhat daunting, diagram. It has been simplified for ease
+of reading with noticeable omissions being a User model, since most models have
+a creator, and the speculative joins of a TimelineEntry.
+
+![Domain model diagram](000x/domain-model-diagram.png)
+
+## Status
+
+Accepted
+
+## Consequences
```

Comment:
I wonder if this section should discuss ideas for migration to the new modelling, as this was definitely a design decision. It does feel somewhat pointless now that it's actually implemented, but it's curious to have so few consequences for such a big proposal; the missing piece seems to be the implementation/migration strategy.